yum.sh

```diff
@@ -6,7 +6,26 @@
 # 
 # Based on apt.sh by Ben Kochie <superq@gmail.com>
 
-upgrades=$(/usr/bin/yum -q check-update | awk 'BEGIN { mute=1 } /Obsoleting Packages/ { mute=0 } mute && /^[[:print:]]+\.[[:print:]]+/ { print $3 }' | sort | uniq -c | awk '{print "yum_upgrades_pending{origin=\""$2"\"} "$1}')
+set -u -o pipefail
+
+filter_awk_script="$(cat << 'AWKSCRIPT'
```

Comment:
Subshell just to capture a multi-line literal?  Why not just:

```
filter_awk_script='stuff
goes
here'
```
or even skip the variable:

```
... check-update |
    awk 'stuff
goes
here' |
   sort | ...

Reply:
I like to use HERE documents in this way to avoid any escaping issues.